### PR TITLE
add Testcontainers

### PR DESCRIPTION
**Pull request by AlekseiKasperovich**

### CODE REVIEW

File: Backend/src/test/java/com/senla/repository/EventRepositoryTest.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.senla.model.Event;
+import com.senla.model.User;
+import java.util.Optional;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+
+class EventRepositoryTest extends DatabaseTest {
```
Comment (vladkondratuk) [inline Backend/src/test/java/com/senla/repository/EventRepositoryTest.java:12]: minor: usually test created for the service layer because there will be the main logic or for the contoller or for both service and controller. 

For the repository layer integration tests need if you have some complex query with joins for example.
But for the most part it's just bacis default CRUD operations that doesn't make much sense to test.

It's fine for the next part I suggest to create integration tests for the service layer.

File: Backend/src/test/java/com/senla/repository/EventRepositoryTest.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+    @Autowired private UserRepository userRepository;
+
+    @Autowired private EventRepository eventRepository;
+
+    @Test
+    void findByAuthorId() {
```
Comment (vladkondratuk) [inline Backend/src/test/java/com/senla/repository/EventRepositoryTest.java:19]: For the test naming I suggest to use concept of GIVEN->WHEN->THEN 

Aslo it's good to have negative case when we can't find author by provided id.

Then test name will be somthing like:
- givenExistingAuthorId_whenFindingById_thenReturnAuthor - for the positive case
- givenNonExistingAuthorId_whenFindingById_thenReturnEmpty - for the negative case

 If this is the service layer then we can throw a NotFoundException so the naming will be;
- givenNonExistingAuthorId_whenFindingById_thenThrowNotFoundException

File: Backend/src/test/java/com/senla/repository/FriendshipRepositoryTest.java [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+
+class FriendshipRepositoryTest extends DatabaseTest {
+
+    @Autowired private UserRepository userRepository;
+
+    @Autowired private FriendshipRepository friendshipRepository;
+
+    @Test
```
Comment (vladkondratuk) [inline Backend/src/test/java/com/senla/repository/FriendshipRepositoryTest.java:18]: You can put spaces between different part of the test. 
Comments can be skiped.
```java
@Test
    void findFriendship() {
        // given
        Optional<User> user1 = userRepository.findByEmail("admin@gmail.com");
        Optional<User> user2 = userRepository.findByEmail("user@gmail.com");
        Friendship friendship =
                Friendship.builder()
                        .sender(user1.get())
                        .receiver(user2.get())
                        .accepted(Boolean.TRUE)
                        .build();
        friendshipRepository.save(friendship);
        
        // when
        Optional<Friendship> foundFriendship =
                friendshipRepository.findFriendship(user1.get().getId(), user2.get().getId()); 
        // then         
        assertThat(foundFriendship.get()).isNotNull();
    }
```

File: Backend/src/test/java/com/senla/repository/FriendshipRepositoryTest.java [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+                        .receiver(user2.get())
+                        .accepted(Boolean.TRUE)
+                        .build();
+        friendshipRepository.save(friendship);
+        Page<Friendship> myFriends = friendshipRepository.findMyFriends(user2.get().getId(), null);
+        assertThat(myFriends.get().findFirst().get()).isNotNull();
+    }
+
+    @Test
+    void getMyFriendshipRequests() {
```
Comment (vladkondratuk) [inline Backend/src/test/java/com/senla/repository/FriendshipRepositoryTest.java:67]: Here also test naming, negative cases.

File: Backend/src/test/java/com/senla/repository/MessageRepositoryTest.java [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+
+class MessageRepositoryTest extends DatabaseTest {
+
+    @Autowired private UserRepository userRepository;
+
+    @Autowired private MessageRepository messageRepository;
+
+    @Test
```
Comment (vladkondratuk) [inline Backend/src/test/java/com/senla/repository/MessageRepositoryTest.java:19]: Here also test naming, negative cases.

File: Backend/src/test/java/com/senla/repository/RoleRepositoryTest.java [hunk 6]
```diff
@@ -5,25 +5,18 @@
...
 
 /** @author Aliaksei Kaspiarovich */
-@RunWith(SpringRunner.class)
-@DataJpaTest
-public class RoleRepositoryTest {
+class RoleRepositoryTest extends DatabaseTest {
 
     @Autowired private RoleRepository roleRepository;
 
     /** Test of findByName method, of class RoleRepository. */
```
Comment (vladkondratuk) [inline Backend/src/test/java/com/senla/repository/RoleRepositoryTest.java:15]: You don't need this comments.

File: Backend/src/test/java/com/senla/service/custom/RoleServiceTest.java [hunk 7]
```diff
@@ -1,36 +1,27 @@
...
 
     @BeforeEach
     public void setUp() {
-        roleRepository = mock(RoleRepository.class);
-        roleService = new RoleServiceImpl(roleRepository);
+        //        roleRepository = mock(RoleRepository.class);
+        //        roleService = new RoleServiceImpl(roleRepository);
     }
 
     /** Test of findByName method, of class RoleService. */
```
Comment (vladkondratuk) [inline Backend/src/test/java/com/senla/service/custom/RoleServiceTest.java:18]: You can remove this comments also.

File: Backend/src/test/java/com/senla/repository/EventRepositoryTest.java [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+        Optional<User> user = userRepository.findByEmail("user@gmail.com");
+        Event event =
+                Event.builder().name("Hello").description("Hello world").author(user.get()).build();
+        eventRepository.save(event);
+        Page<Event> foundEvent = eventRepository.findByAuthorId(user.get().getId(), null);
+        assertThat(foundEvent.get().findFirst().get().getName()).isEqualTo("Hello");
```
Comment (vladkondratuk) [inline Backend/src/test/java/com/senla/repository/EventRepositoryTest.java:25]: I suggest to use junit5 Assertions:

Assertions.assertEquals(foundEvent.get().findFirst().get().getName(), "Hello");

```xml
<dependency>
    <groupId>org.junit.jupiter</groupId>
    <artifactId>junit-jupiter-api</artifactId>
    <version>5.9.0</version>
    <scope>test</scope>
</dependency>
```

File: Backend/src/test/java/com/senla/repository/FriendshipRepositoryTest.java [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+        Friendship friendship =
+                Friendship.builder()
+                        .sender(user1.get())
+                        .receiver(user2.get())
+                        .accepted(Boolean.TRUE)
+                        .build();
+        friendshipRepository.save(friendship);
+        Optional<Friendship> foundFriendship =
+                friendshipRepository.findFriendship(user1.get().getId(), user2.get().getId());
+        assertThat(foundFriendship.get()).isNotNull();
```
Comment (vladkondratuk) [inline Backend/src/test/java/com/senla/repository/FriendshipRepositoryTest.java:31]: Same junit5 -> `Assertions.assertNotNull(foundFriendship.get());`

**Review by vladkondratuk — CHANGES_REQUESTED** [review]
Overoll looks good 👍 

My sugestion: 

- PR naming from `deb ` to `add-testcontainers` can be implemented in the new PRs.
- Skip completly repository layer and proceed with service and controller.
- Use junit5 assertion, upated test naming and add negative test cases.
- You can add controller and service tests in separate PRs.
